core/procedures.js

```diff
@@ -417,7 +417,8 @@ Blockly.Procedures.createProcedureCallbackFactory_ = function(workspace) {
       var blockDom = Blockly.Xml.textToDom(blockText).firstChild;
       Blockly.Events.setGroup(true);
       var block = Blockly.Xml.domToBlock(blockDom, workspace);
-      block.moveBy(30, 30);
+      var scale = workspace.scale; // To convert from pixel units to workspace units
```

Comment:
Add a comment saying that this is putting it at the top left of the visible workspace.

Also, I agree that the right behaviour for RTL is to put it at the top right.  You can test that in the playgrounds by switching to RTL mode and manually calling a function like this to create your block.

If you don't want to do that right now, create an issue describing it and add a comment here that says "TODO(#issue number): RTL)

Reply:
Got it, will add that comment. I'll have a go but might make an issue for the RTL part.